### PR TITLE
Add testRequestTimeout

### PR DESCRIPTION
**Pull request by 8W9aG**
* Tests cases when the request has an absolute
timeout explicitly defined

### CODE REVIEW

**Comment by 8W9aG** [comment]
@jgavris @dflems @rastersize @JensAyton

File: SPTDataLoaderTests/SPTDataLoaderFactoryTest.m [hunk 1]
```diff
@@ -192,4 +195,19 @@ - (void)testRetryAuthorisation
     XCTAssertEqual(requestResponseHandler.numberOfFailedResponseCalls, 1u, @"The factory should only fail once after two authorisation failures");
 }
 
+- (void)testRequestTimeout
+{
+    self.factory.requestTimeoutQueue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0);
+    dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
+    SPTDataLoaderRequestResponseHandlerMock *requestResponseHandler = [SPTDataLoaderRequestResponseHandlerMock new];
+    requestResponseHandler.failedResponseBlock = ^ {
+        dispatch_semaphore_signal(semaphore);
```
Comment (rastersize) [inline SPTDataLoaderTests/SPTDataLoaderFactoryTest.m]: Any reason you’re using a dispatch semaphore over an `XCTestExpectation`?

File: SPTDataLoaderTests/SPTDataLoaderFactoryTest.m [hunk 2]
```diff
@@ -192,4 +195,19 @@ - (void)testRetryAuthorisation
     XCTAssertEqual(requestResponseHandler.numberOfFailedResponseCalls, 1u, @"The factory should only fail once after two authorisation failures");
 }
 
+- (void)testRequestTimeout
+{
+    self.factory.requestTimeoutQueue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0);
+    XCTestExpectation *expectation = [self expectationWithDescription:@"Test timeout executes in time given"];
```
Comment (rastersize) [inline SPTDataLoaderTests/SPTDataLoaderFactoryTest.m]: Add `__weak`?

**Comment by rastersize** [comment]
:+1:

**Pull request merged**